bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -412,7 +424,8 @@ public class TibberWebSocketListener {
         public void onConnect(Session wssession) {
             TibberHandler.this.session = wssession;
             TibberWebSocketListener socket = TibberHandler.this.socket;
-            String connection = "{\"type\":\"connection_init\", \"payload\":\"token=" + tibberConfig.getToken() + "\"}";
+            String connection = "{\"type\":\"connection_init\", \"payload\":{\"token\":\"" + tibberConfig.getToken()
```

Comment:
This looks wild. Any reason to not use GSON here to build the json object and then run toString on it? If the token contains special chars this would break the JSON.

Reply:
Following [Tibber .NET SDK](https://github.com/tibber/Tibber.SDK.NET/commit/4beb5e691b866229f7b4adced64708f0cedada15#diff-43fedf3288bf7eb7404444803d406886bd517a4ed366f890a641fbaec11df5a4R199) it is the messages required for connection initalization and subscription, and it is confirmed to work successfully.

Reply:
If it's valid JSON it would be better for readability and maintainability to build the JSON using GSON. That's just my opinion, maybe the maintainers who review this have a different opinion on that

Reply:
I see your point wrt code "tidiness", but struggle to see why a sudden change in code would be required, potentially also breaking working code, as the "as is" code has been verified working over years. Also taking into account how the API actually is set up.

Reply:
I agree with @Flole998's point that it would be safer to let GSON serialize this into JSON, but since it's not new code, this refactoring is not needed for this PR.